Argo/Globals/Parser.swift

```diff
@@ -0,0 +1,22 @@
+import Box
+
+public enum Parser<T> {
+  case Success(Box<T>)
+  case TypeMismatch(String)
+  case MissingKey(String)
+
+  public var value: T? {
+    switch self {
+    case let .Success(box): return box.value
+    default: return .None
+    }
+  }
+}
+
+func optional<A>(p: Parser<A>) -> Parser<A?> {
```

Comment:
It's internal, so kinda? I do think attaching it to `Parser` still makes sense. The syntax where we're using it won't change much (`optional(foo)` -> `.optional(foo)`)

Given that it's internal, I can let go of this, though.